src/java/org/apache/cassandra/index/sai/plan/TopKProcessor.java

```diff
@@ -75,8 +75,9 @@
 import static java.lang.Math.min;
 
 /**
- * Processor that scans all rows from given partitions and selects rows with top-k scores based on vector indexes.
+ * Processor that scans all rows from given partitions and selects rows with top-k scores based on indexes.
  *
+ * TODO update me based on what I really do
```

Comment:
This class probably needs to be broken up. The post processing use case where we filter results on the coorindator is fairly different than the replica usage of this class, and if you look closely, I think they might not even share that much code. It might be an easy separation.